resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1513,7 +1518,9 @@ private[spark] object Client extends Logging {
     }
 
     sys.env.get(ENV_DIST_CLASSPATH).foreach { cp =>
-      addClasspathEntry(getClusterPath(sparkConf, cp), env)
+      val newCp = cp.split(File.pathSeparator)
+        .filterNot(cpSet.contains).mkString(File.pathSeparator)
```

Comment:
I don't think this is still safe because of classpath order. Let's do this for test only.

Reply:
Ok, hmm... but will `/rootpath/subdir/a-1.0.jar:/rootpath/subdir/b-1.0.jar:/rootpath/subdir/a-1.0.jar` and `/rootpath/subdir/a-1.0.jar:/rootpath/subdir/b-1.0.jar` have different results?

The de-duplication operation is a full path matching